Apps/WebClient/src/ClientApp/src/components/dependentCard.vue

```diff
@@ -0,0 +1,254 @@
+import { ILogger, ILaboratoryService } from "@/services/interfaces";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import User from "@/models/user";
+import { BTabs, BTab } from "bootstrap-vue";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faFileDownload } from "@fortawesome/free-solid-svg-icons";
+library.add(faFileDownload);
+
+@Component({
+    components: {
+        BTabs,
+        BTab,
+    },
+})
+export default class DependentCardComponent extends Vue {
+    @Prop() dependent!: Dependent;
+
+    @Getter("user", { namespace: "user" }) user!: User;
+
+    @Action("getOrders", { namespace: "laboratory" })
+    getLaboratoryOrders!: (params: {
+        hdid: string;
+    }) => Promise<RequestResult<LaboratoryOrder[]>>;
+
+    @Action("addError", { namespace: "errorBanner" })
+    addError!: (error: BannerError) => void;
+
+    private isLoading = false;
+    private logger!: ILogger;
+    private laboratoryService!: ILaboratoryService;
+    private labResults: LaboratoryOrder[] = [];
+    private isDataLoaded = false;
+
+    private get isExpired() {
+        let birthDate = new DateWrapper(this.dependent.dateOfBirth);
+        let now = new DateWrapper();
+        return now.diff(birthDate, "year").years > 19;
+    }
+
+    private mounted() {
+        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+        this.laboratoryService = container.get<ILaboratoryService>(
+            SERVICE_IDENTIFIER.LaboratoryService
+        );
+    }
+
+    private fetchLaboratoryResults() {
+        if (this.isDataLoaded) {
+            return;
+        }
+        this.isLoading = true;
+        this.getLaboratoryOrders({ hdid: this.dependent.hdid })
+            .then((results) => {
+                if (results.resultStatus == ResultType.Success) {
+                    this.labResults = results.resourcePayload;
+                    this.isDataLoaded = true;
+                } else {
+                    this.logger.error(
+                        "Error returned from the laboratory call: " +
+                            JSON.stringify(results.resultError)
+                    );
+                    this.addError(
+                        ErrorTranslator.toBannerError(
+                            "Fetch Laboratory Error",
+                            results.resultError
+                        )
+                    );
+                }
+            })
+            .catch((err) => {
+                this.logger.error(err);
+                this.addError(
+                    ErrorTranslator.toBannerError("Fetch Laboratory Error", err)
+                );
+            })
+            .finally(() => {
+                this.isLoading = false;
+            });
+    }
+
+    private getReport(labOrder: LaboratoryOrder) {
+        let labResult = labOrder.labResults[0];
+        this.laboratoryService
+            .getReportDocument(labResult.id, this.user.hdid)
```

Comment:
should we pass the dependent's hdid or the this.user.hdid here?

Reply:
good point, must be the dependent id!